Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp

```diff
@@ -101,6 +101,9 @@ static const unsigned cMaxLayerTreeDepth = 128;
 // About 10 screens of an iPhone 6 Plus. <rdar://problem/44532782>
 static const unsigned cMaxTotalBackdropFilterArea = 1242 * 2208 * 10;
 
+// Don't let a single tiled layer use more than 128MB of memory.
+static const unsigned cMaxScaledTiledLayerMemorySize = 1024 * 1024 * 128;
```

Comment:
It's unusual to reveal so many tiles on a tiled layers that we hit the contents scale adjustment. The two cases are something large scaled down with transforms, and something rotated in Y or X. When scaled down, the contents scale adjustment compensates in a way that usually still makes the tiles rendered at higher resolution than necessary, so blurriness will be rare, I think. I chose 128MB because the Apple Pro XDR display is 6016 * 3384 which is about 78MB.

Reply:
I chose 156MB for this patch, based on wanting 12 3x tiles on an iPhone display.